src/Domain/Entity/Network.php

```diff
@@ -173,4 +178,14 @@ public function isRadio(): bool
     {
         return ($this->getMedium() === NetworkMediumEnum::RADIO);
     }
+
+    public function getOptions(): Options
```

Comment:
Options is the superclass of UnfetchedOptions. So the return type (generic) is Options as can be both. In this case UnfetchedOptions has only an empty array of options.

This said is the responsibility ProgrammeMapper to figure out if the whole hierarchy has been fetched. Otherwise the programme will have an UnfetchedOptions.

In the other side NetworkMapper doesn't need to check the hierarchy, as they are in the top hierarchy. So they set always Options.

Sometimes is also possible to have (before this new feature) to get an UnfetchedNetwork, UnfetchedEpisode, etc. For all of these Unfetched entities we set also UnfetchedOptions.